web/services/lib/filters/filters.ts

```diff
@@ -173,3 +176,44 @@ export function buildFilter(
     argsAcc: res.argsAcc,
   };
 }
+
+async function getUserIdHashes(user_id: string): Promise<string[]> {
+  const { data: user_api_keys, error } = await supabaseServer
+    .from("user_api_keys")
+    .select("api_key_hash")
+    .eq("user_id", user_id);
+  if (error) {
+    throw error;
+  }
+  if (!user_api_keys || user_api_keys.length === 0) {
+    throw new Error("No API keys found for user");
```

Comment:
Why is this a throw? Isn't it feasible that a user has no api keys?

E.g., I deleted all my keys

Reply:
Then you have no data anyway if you delete all your keys. I wanted to throw because I did not want them to get into a state where we would accidentally return ALL data without a filter